cloudmarker/stores/mongodbstore.py

```diff
@@ -40,6 +40,27 @@ def __init__(self, db, username, password, host, port=27017,
 
         self._db = self._client[db]
 
+        # Create collections if they don't exist based on the document
+        # restrictions the users want to have
+        """The collection creation, where the models objects specified, are
+        done with the following way. The object specified must have the
+        following functions:
+            collection(): returns the name of the collection
+            validator(): returns a valid string with $jsonSchema that contains
+                all the validation properties as specified in the mongodb
+                docs.
+            enforce(): returns one of the strings ``error`` or ``warn``.
+                This will be used to define the error level for when the schema
+                validation fails. ``error`` will raise an error while ``warn``
+                will log that violation in the db server logs.
+        """
+        for model in kwargs.get('models', []):
+            if model.collection() not in self._db.list_collection_names():
+                self._db.create_collection(
+                    model.collection(),
+                    validator=model.validator(),
+                    validationAction=model.enforce()
+                )
```

Comment:
If it turns out that this change should be merged to `master`, please update the commit message to:

    Add optional validations for MongoDB schema

In general, the commit messages need to be very short (not exceeding 50 characters in length) but very precise with just enough detail to tell us what exactly is going on in the change without compromising on the way we write brand names ("MongoDB" in this case).